Sources/eppo/UniversalFlagConfig.swift

```diff
@@ -0,0 +1,84 @@
+import Foundation
+
+public struct UniversalFlagConfig : Decodable {
+    let createdAt: Date?;
+    let flags: [String: UFC_Flag];
+    // todo: add bandits
+    
+    static func decodeFromJSON(from jsonString: String) throws -> UniversalFlagConfig {
+        let decoder = JSONDecoder()
+
+        // Set up the date formatter
+        let dateFormatter = DateFormatter()
+        dateFormatter.dateFormat = "yyyy-MM-dd'T'HH:mm:ss.SSSZ"  // Adjusted to include milliseconds
+        dateFormatter.locale = Locale(identifier: "en_US_POSIX")  // Use POSIX to avoid unexpected behaviors
+        dateFormatter.timeZone = TimeZone(secondsFromGMT: 0)  // Adjust if your JSON dates are not in GMT
+
+        // Use the date formatter in the decoder
+        decoder.dateDecodingStrategy = .formatted(dateFormatter)
+
+        guard let jsonData = jsonString.data(using: .utf8) else {
+            throw ConfigurationRequesterError.invalidJSON("Cannot be encoded into UTF-8")
+        }
+
+        // Attempt to validate JSON structure
+        do {
+            _ = try JSONSerialization.jsonObject(with: jsonData, options: [])
+        } catch let error {
+            throw ConfigurationRequesterError.invalidJSON("Invalid JSON: \(error.localizedDescription)")
+        }
+
+        // Attempt to decode the JSON into RACConfig
+        return try decoder.decode(UniversalFlagConfig.self, from: jsonData)
+    }
+}
+
+public struct UFC_Flag : Decodable {
+    let key: String;
+    let enabled: Bool;
+    let variationType: String; // todo: enum
+    let variations: [String: UFC_Variation];
+    let allocations: [UFC_Allocation];
+    let totalShards: Int;
+}
+
+public struct UFC_Variation : Decodable  {
+    let key: String;
+    let value: EppoValue;
+    let algorithmType: String? // todo: enum
+}
+
+public struct UFC_Allocation : Decodable {
+    let key: String;
+    let rules: [UFC_Rule]?;
```

Comment:
I kind of wish this was required - seems like it invites room for error. Should be fine here; some of the test files have this missing and unit tests would catch it.